src/Microsoft.AspNetCore.Hosting/Internal/RequestServicesContainerMiddleware.cs

```diff
@@ -30,34 +30,30 @@ public RequestServicesContainerMiddleware(RequestDelegate next, IServiceScopeFac
             _scopeFactory = scopeFactory;
         }
 
-        public async Task Invoke(HttpContext httpContext)
+        public Task Invoke(HttpContext httpContext)
         {
             Debug.Assert(httpContext != null);
 
             // local cache for virtual disptach result
             var features = httpContext.Features;
-            var existingFeature = features.Get<IServiceProvidersFeature>();
 
-            // All done if RequestServices is set
-            if (existingFeature?.RequestServices != null)
-            {
-                await _next.Invoke(httpContext);
-                return;
-            }
+            var replacementFeature = new RequestServicesFeature(httpContext, _scopeFactory);
 
-            var replacementFeature = new RequestServicesFeature(_scopeFactory);
+            features.Set<IServiceProvidersFeature>(replacementFeature);
+            var task = _next.Invoke(httpContext);
```

Comment:
If you're not going to do any cleanup afterwards then why not return immediately here?

Reply:
Yea this is true. I was shuffling the code so much before I don't even need the other part.